src/web/page/web_page.rs

```diff
@@ -0,0 +1,57 @@
+use super::TEMPLATE_DATA;
+use iron::{headers::ContentType, response::Response, status::Status, IronResult};
+use serde::Serialize;
+use tera::Context;
+
+#[macro_export]
+macro_rules! impl_webpage {
+    ($page:ty = $template:expr $(, status = $status:expr)? $(, content_type = $content_type:expr)? $(,)?) => {
+        impl $crate::web::page::WebPage for $page {
+            const TEMPLATE: &'static str = $template;
+
+            $(
+                fn get_status(&self) -> ::iron::status::Status {
+                    $status
+                }
+            )?
+
+            $(
+                fn content_type() -> ::iron::headers::ContentType {
+                    $content_type
+                }
+            )?
```

Comment:
Both these functions look like they could be associated constants instead, which would decrease the boilerplate a lot and possibly get rid of the need for a macro.

Reply:
They can’t be, status can be gotten from the template struct itself and is, while not all content types are statically constructible

Reply:
> not all content types are statically constructible

This seems strange to me, can you give an example?

Reply:
I imagine the content type of rustdoc pages is not only HTML, but also CSS and JS.

Reply:
[`ContentType::html()`](https://docs.rs/iron/0.6.1/iron/headers/struct.ContentType.html#method.html) isn’t a `const fn`